app/services/user_service.rb

```diff
@@ -1,4 +1,13 @@
 class UserService
+  def self.destroy(user:)
+    user.deactivate!
+    zombie = User.create(name: I18n.t(:'user.deleted_user'),
+                         email: "deleted-user-#{rand(10**10)}@example.com")
+    MigrateUserService.migrate!(source: user, destination: zombie)
+    user.reload.destroy
+    zombie
```

Comment:
I don't believe we need to return the zombie; if we wanted to serialize it out we'd have to do
```
Events::UserDestroyed.publish!(user, zombie)
```
and have the zombie be the eventable for the event that's created.

(But I don't think we need that.)

Reply:
Also, we should chuck a
```
EventBus.broadcast 'user_destroy', user, zombie
```
on there for completeness.